Mathlib/Algebra/Star/Subalgebra.lean

```diff
@@ -107,8 +107,7 @@ theorem toSubalgebra_le_iff {S₁ S₂ : StarSubalgebra R A} :
 equalities. -/
 protected def copy (S : StarSubalgebra R A) (s : Set A) (hs : s = ↑S) : StarSubalgebra R A where
   toSubalgebra := Subalgebra.copy S.toSubalgebra s hs
-  star_mem' := @fun a ha => hs ▸ (S.star_mem' (by simpa [hs] using ha) : star a ∈ (S : Set A))
-  -- Porting note: the old proof kept crashing Lean
+  star_mem' {a} ha := hs ▸ S.star_mem' (by simpa [hs] using ha)
```

Comment:
I'm not sure what the old proof was, but this one seems fine.

Reply:
The Mathlib 3 proof was `λ _, hs.symm ▸ S.star_mem'`, which needs slight tweaks in Lean 4 because the triangle doesn't seem to rewrite the `ha` argument. So imo the `simpa` is fine here.